data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/PostStateValidatorBalancesIntegrationTest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Consensys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.beaconrestapi.v1.beacon;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+
+import okhttp3.Response;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.beaconrestapi.AbstractDataBackedRestAPIIntegrationTest;
+import tech.pegasys.teku.beaconrestapi.handlers.v1.beacon.GetStateValidatorBalances;
+
+public class PostStateValidatorBalancesIntegrationTest
```

Comment:
Where are the success cases? Including filtering and returning all?

Reply:
The unit tests cover more scenarios, do we need to retest this as an integration test?

Reply:
This is testing bad injected json which we can't easily replicate in a unit test (its the response 400 we're verifying rather than 500). I think we're good here...

Reply:
I was thinking of the handling of empty request body that isn't exercised in the unit test.

Reply:
yep that'd make sense too...

Reply:
I did test that in the unit tests, maybe I will rename the test so it is clearer